File: 01BouncingBall/01.2Swap/Main.java
import java.util.Scanner;

public class Main {

    public static void swap(int a, int b){

        a = a - b;
        b = a + b;
        a = b - a;

    }

	public static void main(String[] args){
    
    Scanner sc = new Scanner(System.in); //used to retrive user input
    int a,b;

    System.out.print("a = "); a = sc.nextInt();
    System.out.print("b = "); b = sc.nextInt();

    System.out.println("a = " + a + " b = " + b + "\nSWAP");
    swap(a,b);
    System.out.println("a = " + a + " b = " + b);
        
    }
}

File: 01BouncingBall/01.2Swap/main.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

#This small program swaps two variables without a temp variable
#Made by Attila Fekete at the University of Debrecen in 2018

def swap(a,b):
    
    a = a - b
    b = a + b
    a = b - a
    return a,b

#The program stars here

a = input("a = ")
b = input("b = ")

print "a =" , a , " b =" , b , "\nSWAP"
a,b = swap(a,b)
print "a =" , a , " b =" , b
